src/main/java/org/broadinstitute/hellbender/utils/MathUtils.java

```diff
@@ -674,22 +608,14 @@ public static double log10SumLog10(final double[] log10Values, final int start,
 
         // we may decide to just normalize in log space without converting to linear space
         if (keepInLogSpace) {
-            for (int i = 0; i < array.length; i++) {
-                array[i] -= maxValue;
-            }
-            return array;
+            return Arrays.stream(array).map(x -> x - maxValue).toArray();
```

Comment:
I wrote a short method in `MathUtils` to apply a lambda to an array without creating a stream.  We get a compact one-liner with the speed of pure arrays.  I benchmarked this and it gets the same performance as arrays; both are much better than streams.  Even though this new method takes in a `Function<Double, Double>` it appears the JIT compiler is smart enough to realize when to skip the boxing and unboxing.